ai2thor/video_controller.py

```diff
@@ -0,0 +1,346 @@
+           step(action='ToggleMapView') directly, since it allows for automatic
+           ceiling toggles in the future. (e.g., if the camera is above the
+           height of the room, toggle off the ceiling, and vice versa."""
+        self.ceiling_off = not self.ceiling_off
+        return self.step(action='ToggleMapView')
+
+    def _cdf(self, x, std_dev=0.5, mean=0.0):
+        '''Cumulative distribution function'''
+        return (1.0 + erf((x - mean) / sqrt(2.0 * std_dev**2))) / 2.0
+
+    def _linear_to_smooth(self, curr_frame, total_frames, std_dev=0.5, min_val=3):
+        # start at -3 STD on a normal gaussian, go to 3 STD on gaussian
+        # curr frame should be 1 indexed, and end with total_frames
+        assert min_val > 0, "Min val should be > 0"
+
+        if curr_frame == total_frames:
+            # removes drifting
+            return 1
+
+        return self._cdf(- min_val + 2 * min_val * (curr_frame / total_frames), std_dev=std_dev)
+
+    def _move(self, actionName, moveMagnitude, frames, smoothAnimation, agentId=None):
+        """Yields a generator full of move commands to move the agent incrementally.
+           Used as a general move command for MoveAhead, MoveRight, MoveLeft, MoveBack."""
+        last_moveMag = 0
+        for i in range(frames):
+            # smoothAnimation = False => linear animation
+            if smoothAnimation:
+                next_moveMag = self._linear_to_smooth(i + 1, frames, std_dev=1) * moveMagnitude
+                if agentId == None:
+                    yield self.step(action=actionName, moveMagnitude=next_moveMag - last_moveMag)
+                else:
+                    yield self.step(action=actionName, moveMagnitude=next_moveMag - last_moveMag, agentId=agentId)
+                last_moveMag = next_moveMag
+            else:
+                if agentId == None:
+                    yield self.step(action=actionName, moveMagnitude=moveMagnitude / frames)
+                else:
+                    yield self.step(action=actionName, moveMagnitude=moveMagnitude / frames, agentId=agentId)
+
+    def _rotate(self, direction, rotateDegrees, frames, smoothAnimation, agentId=None):
+        """Yields a generator full of step(action='TeleportFull') commands to rotate the agent incrementally."""
+        if agentId is not None:
+            raise ValueError('rotations do not yet work with multiple agents')
+
+        # make it work for left and right rotations
+        direction = direction.lower()
+        assert direction == 'left' or direction == 'right'
+        if direction == 'left': rotateDegrees *= -1
+
+        # get the initial rotation
+        y0 = self.last_event.metadata['agent']['rotation']['y']
+        for i in range(frames):
+            # keep the position the same
+            p = self.last_event.metadata['agent']['position']
+            if smoothAnimation:
+                if agentId == None:
```

Comment:
All == None should be changed to 'is None'

```python
if agentId is None
```

See https://stackoverflow.com/questions/14247373/python-none-comparison-should-i-use-is-or